clients/src/main/java/org/apache/kafka/common/requests/OffsetCommitResponse.java

```diff
@@ -167,8 +213,24 @@ public <P> Builder addPartitions(
         }
 
         public Builder merge(
-            OffsetCommitResponseData newData
+            OffsetCommitResponseData newData,
+            Logger logger
         ) {
+            if (version >= 9) {
+                // This method is called after the group coordinator committed the offsets. The group coordinator
+                // provides the OffsetCommitResponseData it built in the process. As of now, this data does
+                // not contain topic ids, so we resolve them here.
+                newData.topics().forEach(topic -> {
+                    Uuid topicId = topicResolver.getTopicId(topic.name()).orElse(Uuid.ZERO_UUID);
+                    if (Uuid.ZERO_UUID.equals(topicId)) {
+                        // This should not happen because topic names returned by the group coordinator should
+                        // always be resolvable.
+                        logger.debug("Unresolvable topic id for topic {} while preparing " +
```

Comment:
Hmm, we probably don't want to include a topic without id in the response version 9 here.